packages/core/src/lib/Cache.ts

```diff
@@ -4,158 +4,121 @@ import { ICategory } from "./Category.js"
 import { trace } from "@opentelemetry/api"
 import { Tags } from "../index.js"
 
-type Expiration = { absolute: number } | { relative: number }
-
-export class CacheEntry<State> {
+export class CacheEntry<T> {
   constructor(
     public token: StreamToken,
-    public state: State,
+    public state: T,
+    public cachedAt: number,
   ) {}
 
-  updateIfNewer(supersedes: (a: StreamToken, b: StreamToken) => boolean, other: CacheEntry<State>) {
+  updateIfNewer(other: CacheEntry<T>) {
     if (supersedes(this.token, other.token)) {
       this.token = other.token
       this.state = other.state
+      this.cachedAt = Date.now()
     }
   }
-  value(): TokenAndState<State> {
+  value(): TokenAndState<T> {
     return { token: this.token, state: this.state }
   }
 
   static ofTokenAndState<S>(x: TokenAndState<S>) {
-    return new CacheEntry(x.token, x.state)
+    return new CacheEntry(x.token, x.state, Date.now())
```

Comment:
is this def the most efficient way to compute an instant? the actual date is not of relevance and e.g. we are not interested in daylight savings or clock smoothing etc (actually thinking of how heavy a date API can be triggerd by a recollection of it dominating a profile in C++!)

Reply:
`Date.now()` returns a unix timestamp in ms. I'll run some micro benchmarks but pretty sure it's fine™️